src/capture.rs

```diff
@@ -518,3 +527,32 @@ mod serde_impls {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_frame_conversion() {
+        // captures an original backtrace, and makes sure that the manual conversion
+        // to frames yields the same results.
+        let bt = Backtrace::new();
+        let original_frames = bt.frames();
+        let mut original = original_frames.iter();
+
+        let mut frames = 0;
+        unsafe {
+            crate::trace_unsynchronized(|frame| {
```

Comment:
Could this use `trace` instead of `trace_unsynchronized`? Additionally can this use the `BacktraceFrame` after the closure to ensure that it's not accidentally holding on to any ephemeral pointeres?

Reply:
yeah, we can use `trace`. I ended up using the unsynchronized version because that is what I was using in my code, no problem.

By using the backtrace frame, maybe I can return a vector and do the asserts by zipping both iterators. Would that be enough for you ?

Reply:
Thanks for your comments. I also noticed that the test would fail in `--release` mode, and indeed skipping a frame is fragile as inlining can change it.

I used a better method that now succeeds with release and debug.

A new version is force-pushed.

Reply:
Whatever testing is fine by me,  I mainly just want to confirm that if the `BacktraceFrame` leaks out of the closure then it's still safe to use.

Reply:
Ok, just saw that. I bet it is a similar issue with `--release` and `--debug`. I will try it locally and get back to you

Reply:
Ok. The issue with miri is different: every backtrace generated has completely different addresses!
Fortunately they resolve to the same thing, so I am testing with that

Reply:
my god, lol...

At least your code coverage is pretty good!!

The test that fails now is about features disabled. If I understand your code correctly, that's because resolving symbols need features to run. I can remove the last assert. It won't be doing that much in that case, but it is better than disabling the whole test in the absence of features. At least the conversion path is stressed.